newlib/libc/include/stdint.h

```diff
@@ -457,6 +457,17 @@ typedef __uint_least64_t uint_least64_t;
 #endif
 #endif
 
-_END_STD_C
+#if __STDC_WANT_LIB_EXT1__ == 1
+#include <sys/_types.h>
 
-#endif /* _STDINT_H */
+  // could be defined by the user
+#ifndef RSIZE_MAX
+#define RSIZE_MAX SIZE_MAX
+#endif
+
+  typedef __rsize_t rsize_t;
```

Comment:
Like `errno_t`, `rsize_t` must be defined from multiple files: stddef.h, stdio.h, stdlib.h, string.h, time.h and wchar.h. Use the same pattern as `errno_t`:
```
#ifndef _RSIZE_T_DEFINED
typedef __rsize_t rsize_t;
#define _RSIZE_T_DEFINED
#endif
```